compiler/src/dotty/tools/dotc/transform/TailRec.scala

```diff
@@ -174,12 +166,51 @@ class TailRec extends MiniPhase {
 
   }
 
-  class TailRecElimination(method: Symbol, enclosingClass: Symbol, isMandatory: Boolean, label: Symbol) extends tpd.TreeMap {
+  class TailRecElimination(method: Symbol, enclosingClass: Symbol, paramSyms: List[Symbol], isMandatory: Boolean) extends tpd.TreeMap {
 
     import dotty.tools.dotc.ast.tpd._
 
     var rewrote: Boolean = false
 
+    var continueLabel: Option[Symbol] = None
+    var varForRewrittenThis: Option[Symbol] = None
+    var rewrittenParamSyms: List[Symbol] = Nil
+    var varsForRewrittenParamSyms: List[Symbol] = Nil
```

Comment:
No, `rewrittenParamSyms` is a *subset* of `paramSyms`. It only contains parameters whose value is changed in at least one tail-recursive call. This is quite important to avoid declaring and assigning useless variables.